providers/gov/nih/normalizer.py

```diff
@@ -215,14 +281,52 @@ def maybe_org(self, obj):
 
     def get_organization_ctx(self, ctx):
         org_ctx = {
-            'ORG_NAME': ctx['ORG_NAME'],
-            'ORG_FIPS': ctx['ORG_FIPS'],
-            'ORG_DEPT': ctx['ORG_DEPT'],
-            'ORG_DUNS': ctx['ORG_DUNS'],
-            'ORG_DISTRICT': ctx['ORG_DISTRICT'],
-            'ORG_CITY': ctx['ORG_CITY'],
-            'ORG_STATE': ctx['ORG_STATE'],
-            'ORG_ZIPCODE': ctx['ORG_ZIPCODE'],
-            'ORG_COUNTRY': ctx['ORG_COUNTRY']
+            'ORG_NAME': filter_nil(ctx['ORG_NAME']),
+            'ORG_FIPS': filter_nil(ctx['ORG_FIPS']),
+            'ORG_DEPT': filter_nil(ctx['ORG_DEPT']),
+            'ORG_DUNS': filter_nil(ctx['ORG_DUNS']),
+            'ORG_DISTRICT': filter_nil(ctx['ORG_DISTRICT']),
+            'ORG_CITY': filter_nil(ctx['ORG_CITY']),
+            'ORG_STATE': filter_nil(ctx['ORG_STATE']),
+            'ORG_ZIPCODE': filter_nil(ctx['ORG_ZIPCODE']),
+            'ORG_COUNTRY': filter_nil(ctx['ORG_COUNTRY'])
         }
         return org_ctx
+
+    def get_pi(self, ctx, primary=True):
+        '''
+            <PI>
+                <PI_NAME>VIDAL, MARC  (contact)</PI_NAME>
+                <PI_ID>2094159 (contact)</PI_ID>
+            </PI>
+        '''
+        pi_list = ctx['PIS']['PI'] if isinstance(ctx['PIS']['PI'], list) else [ctx['PIS']['PI']]
+        org_ctx = self.get_organization_ctx(ctx)
+        # only one or none so primary is that one
+        if len(pi_list) <= 1 and primary:
+            pi_list[0]['org_ctx'] = org_ctx
+            return pi_list
+        # more than one, get the primary
+        if len(pi_list) > 1 and primary:
+            pi_itr = iter(pi_list)
+            while True:
+                try:
+                    pi = next(pi_itr)
+                except StopIteration:
+                    return []
+                if '(contact)' in pi['PI_NAME']:
+                    return {
+                        'PI_NAME': re.sub(r'(\(contact\))', '', pi['PI_NAME']).strip(),
+                        'PI_ID': re.sub(r'(\(contact\))', '', pi['PI_ID']).strip(),
+                        'org_ctx': org_ctx
+                    }
+        # more than one, get the non-primary
+        if len(pi_list) > 1 and not primary:
+            non_primary_pi = []
+            for pi in pi_list:
+                if '(contact)' not in pi['PI_NAME']:
+                    pi['org_ctx'] = org_ctx
+                    non_primary_pi.append(pi)
+            return non_primary_pi
+        # only one and not primary
+        return None
```

Comment:
It might be a bit more clear if you hoist this logic to the beginning of these if blocks

```
if len(x) <= 1:
    if not primary:
        return None
    # logic
    return value

if primary
    # logic
    return value

# logic
return logic
```